onnxruntime/core/graph/model.cc

```diff
@@ -82,10 +82,17 @@ Model::Model(const std::string& graph_name,
     opset_id_proto->set_version(domain.second);
   }
 
+  std::vector<const ONNX_NAMESPACE::FunctionProto*> model_functions;
+  for (auto& func : model_local_functions) {
+    auto func_ptr = model_proto_.add_functions();
+    func_ptr->CopyFrom(func);
+    model_functions.emplace_back(func_ptr);
+  }
```

Comment:
nit: prefer push_back [https://abseil.io/tips/112](https://abseil.io/tips/112)

would also be nice to call reserve on model_functions.